src/integration/sink/tls.go

```diff
@@ -0,0 +1,43 @@
+package sink
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/newrelic/nri-kubernetes/v3/internal/config"
+)
+
+var ErrCAAppend = errors.New("appending certs to pool")
+
+func NewTLSClient(conf config.TLSConfig) (*http.Client, error) {
```

Comment:
I don't see as a bad approach to create a NewTLSClient_test that ensures that it can make calls to a mockedServer :
` srv := NewServer(httpsPort)
  go srv.ListenAndServeTLS("cert.pem", "key.pem")
  defer srv.Close()`

Reply:
Because at the end, you are not testing http.Client but you are ensuring that your constructor creates a valid httpClient.

Reply:
This does make a lot of sense, and it's perfectly testable. I'll add a test for it :)

Reply:
The main problem with this is how incredibly painful it is to create an HTTPs server for testing. `httptest` uses hardcoded certificates, and to create an "standard" one you need to add a _lot_ of (potentially buggy) boilerplate to set it up.

I'll give it a try but I think it will be awfully complex.